### DISCUSSION TITLE
Typescript generics :-)

### DISCUSSION THREAD

**Discussion by iangregsondev** [ben-rogerson/twin.macro#265] [General]
Going to throw something strange out there :-)

I just found this library by mistake and its going to be really helpful!, I also wanted the ability to grab the values of the classes so I can use it in my code for animation etc...

Anyway ... It would be great if you could pass in interface into generics i.e.

Currently

```
// input
const test = ${tw`text-center bg-red w-full`}

// output
const test = {
    background: 'red',
    textAlign: 'center',
    width: '100%'
}
```

Adding generics into the mix would be able to give typescript code completion.

```
interface MyStyle {
    background: string,
    textAlign:  "left" | "center" | "right",
    width: string
}

//Pass interface as part of generic, not sure about the syntax here :-(

const test = ${tw<MyStyle>`text-center bg-red w-full`}

// Now you have types and code completion i.e.

console.log(test.background)

if (test.textAligh == "left") {
// etc
}
```

Not sure if this is something that would be possible, it sure would be sweet :-)

**Comment by ben-rogerson** [comment]
This could be worth looking into. If you (or anyone reading this) would like to help add this feature, please sing out.
